README.md

```diff
@@ -1,58 +1,538 @@
 
+java -javaagent:/[user path]/easeagent.jar=/{path}/agent.properties -Deaseagent.server.enabled=false -jar spring-petclinic-api-gateway/target/spring-petclinic-api-gateway-2.4.2.jar
 ```
-java -jar -javaagent:/{path}/easeagent.jar=/{path}/agent.properties" -Deaseagent.log.conf=/{path}/log4j2.xml -Deaseagent.server.port=9901 your-springboot-app.jar
+
+### Step 5
+Tracing and Metric information can be shown in console  
+
+## Documentation
+
+### Agent Configuration
+
+### Health Check Endpoint
+User can use the following URL to support health check and liveness check
+```
+http://[ip]:[easeagent.server.port]/health
 ```
-## Other Configuration Detail
-Detailed configuration information is in build/src/main/resources/agent.properties
 
-# Build From Source
+### Readiness Check Endpoint
+User can use the following URL to support health check in SpringBoot 2.2.x
+```
+http://[ip]:[easeagent.server.port]/health/readiness
+```
+
+### Tracing
+EaseAgent use [brave](https://github.com/openzipkin/brave) to collect tracing information.The data format stored in `Kafka`  is [Zipkin Data Model](https://zipkin.io/pages/data_model.html). User can send tracing information to [Zipkin server](https://zipkin.io/pages/quickstart.html).
+
+### Metrics
+
+EaseAgent use `io.dropwizard.metrics` to collect metric information.
+The metric name's rule are as follows:
+
+ Metric | Metric Type| 
+---|---|
+Timers | 0
+Histograms | 1
+Meters | 2
+Counters | 3
+Gauges | 4
+
+ Key Type | Metric Sub Type| Description |
+---|---|---|
+SUCCESS | 00 | Method invoke d not exception 
+ERROR | 01 | Method call has error or http response 500, etc
+CHANNEL | 02 | RabbitMq publish message to channel or queue
+CONSUMER | 03 | RabbitMq or Kafka consume message
+PRODUCER | 04 | Kafka send message to topic
+CONSUMER_ERROR | 05 | RabbitMq or Kafka consume message has error
+PRODUCER_ERROR | 06 | Kafka send message has error
 
 ```
-mvn clean package -am -pl build
+The final key is [Metric Sub Type] + [Metric Type] + subKey
+```
+
+For Example:
+
+User send message to Kafka topic `demo-topic`. 
+```
+subkey = "demo-topic"
+The producer's Counters metric key for for success is [04]+[3]+[demo-topic] = 043demo-topic 
+The producer's Counters metric key for for error is [06]+[3]+[demo-topic] = 063demo-topic 
+```
+
+#### Prometheus Metrics Support
+EaseAgent support Prometheus pull metrics info from the following configuration
+```
+  - job_name: 'user-app'
+    static_configs:
+    - targets: ['ip:easeagent.server.port']
+    metrics_path: "/prometheus/metrics"
+```
+Please note: Prometheus Label names must match the regex `[a-zA-Z_][a-zA-Z0-9_]*`. The character `-` will be replaced with `_`.
+
+See [Prometheus Metric names and labels rule](https://prometheus.io/docs/concepts/data_model)
+
+#### Metric Field
```

Comment:
The section of metric can be omitted in README.md, It should be written in the manual documents